lib/Decompressor.js

```diff
@@ -32,20 +34,61 @@ export class Decompressor extends Transformer {
    * compressed CBOR-LD. The created instance may only be used on a single
    * CBOR-LD input at a time.
    *
-   * @param {object} options - The options to use when encoding to CBOR-LD.
-   * @param {documentLoaderFunction} options.documentLoader - The document
+   * @param {object} options - The options to use.
+   * @param {documentLoaderFunction} options.documentLoader -The document
    *   loader to use when resolving JSON-LD Context URLs.
-   * @param {Map} [options.appContextMap] - A map of JSON-LD Context URLs and
-   *   their encoded CBOR-LD values (must be values greater than 32767
-   *   (0x7FFF)).
+   * @param {Map} options.keywordsTable -A map of JSON-LD keywords
+   *  and their associated CBOR-LD integer values.
+   * @param {Map} options.stringTable - A map of string values and
+   *  their associated CBOR-LD integer values.
+   * @param {Map} options.urlSchemeTable - A map of URL schemes to
+   *  their associated CBOR-LD integer values.
+   * @param {Map} options.typedLiteralTable - A map of JSON-LD types
+   *  to maps that map values of that type to their associated CBOR-LD
+   *  integer values.
    */
-  constructor({documentLoader, appContextMap} = {}) {
-    super({documentLoader, appContextMap});
-    this.reverseAppContextMap = new Map();
-    // build reverse contxt map
-    if(appContextMap) {
-      for(const [k, v] of appContextMap) {
-        this.reverseAppContextMap.set(v, k);
+  constructor({
+    documentLoader,
+    keywordsTable,
+    urlSchemeTable,
+    stringTable,
+    typedLiteralTable
+  } = {}) {
+    super({
+      documentLoader,
+      keywordsTable,
+      urlSchemeTable,
+      stringTable,
+      typedLiteralTable
+    });
+    this.idToTerm = new Map();
+    this.reverseStringTable = new Map();
+    this.reverseUrlSchemeTable = new Map();
+    this.reverseTypedLiteralTable = new Map();
+    // build reverse compression tables
+    if(keywordsTable) {
+      for(const [k, v] of keywordsTable) {
+        this.idToTerm.set(v, k);
+      }
+    }
+    if(stringTable) {
+      for(const [k, v] of stringTable) {
+        this.reverseStringTable.set(v, k);
+      }
+    }
+    if(urlSchemeTable) {
+      for(const [k, v] of urlSchemeTable) {
+        this.reverseUrlSchemeTable.set(v, k);
+      }
+    }
+    //typed literal table is map of maps, process inner map not outer
+    if(typedLiteralTable) {
+      for(const [k, v] of typedLiteralTable) {
+        const tempReverseMap = new Map();
+        for(const [k2, v2] of v) {
+          tempReverseMap.set(v2, k2);
+        }
+        this.reverseTypedLiteralTable.set(k, tempReverseMap);
       }
     }
```

Comment:
I *think* this might work and be shorter:

```suggestion
    // build reverse compression tables
    this.idToTerm = new Map(
      Array.from(keywordsTable, e => e.reverse()));
    this.reverseStringTable = new Map(
      Array.from(reverseStringTable, e => e.reverse()));
    this.reverseUrlSchemeTable = new Map(
      Array.from(reverseUrlSchemeTable, e => e.reverse()));
    // typed literal table is map of maps, process inner maps not outer
    this.reverseTypedLiteralTable = new Map();
    if(typedLiteralTable) {
      for(const [k, map] of typedLiteralTable) {
        this.reverseTypedLiteralTable.set(
          k, new Map(Array.from(map, e => e.reverse())));
      }
    }
```

Reply:
I'd recommend adding a little `_reverseMap(m)` helper that can be called:

```js
function _reverseMap(m) {
  return new Map(Array.from(m, e => e.reverse());
}
```